### PR TITLE
add POST to ratings service to demonstrate security policies on HTTP Methods

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
To allow GET and forbid POST methods, to demonstrate the difference between Istio security policies and Kubernetes Network policies. While Kubernetes Network policies can define which pod can call which pod, they cannot define which methods can be applied, for example, to allow pod A send only GET requests to pod B, and to allow pod C send both GET AND POST requests to pod B.

### CODE REVIEW

**Comment by frankbu** [comment]
Is this going to be used in a new task about Istio security policies? If so wouldn't it be simpler to just use the sleep service and httpbin for the new task instead of adding extra code to the bookinfo sample?

**Comment by dkrieger** [comment]
Is it currently (stable: 1.0.5) possible to block POST requests to some service, or is this new in 1.1? I looked at [the docs](https://istio.io/docs/tasks/traffic-management/egress/) for an answer and couldn't find any indication.

**Comment by vadimeisenbergibm** [comment]
@dkrieger You can define Mixer policies based on the attributes specified in https://istio.io/docs/reference/config/policy-and-telemetry/attribute-vocabulary/. In particular on `request.method`. This capability has always been there. For Egress traffic you have to direct it thru egress gateway to apply policies, see https://archive.istio.io/v0.8/blog/2018/egress-monitoring-access-control/.

File: samples/bookinfo/src/ratings/ratings.js [hunk 1]
```diff
@@ -118,7 +148,18 @@ dispatcher.onGet('/health', function (req, res) {
...
+function putLocalReviews (productId, ratings) {
+  userAddedRatings[productId] = {
+    id: productId,
+    ratings: ratings
+  }
+  return getLocalReviews(productId)
+}
+
 function getLocalReviews (productId) {
+  if(typeof userAddedRatings[productId] !== 'undefined') {
```
Comment (venilnoronha) [inline samples/bookinfo/src/ratings/ratings.js]: nit: `if(` -> `if (`

File: samples/bookinfo/src/ratings/ratings.js [hunk 2]
```diff
@@ -33,6 +35,34 @@ if (process.env.SERVICE_VERSION === 'v2') {
...
+
+  try {
+    ratings = JSON.parse(req.body)
+  } catch (error) {
+    res.writeHead(400, {'Content-type': 'application/json'})
+    res.end(JSON.stringify({error: 'please provide valid ratings JSON'}))
+    return
+  }
+
+  if (process.env.SERVICE_VERSION === 'v2') {
```
Comment (venilnoronha) [inline samples/bookinfo/src/ratings/ratings.js]: I'd add a comment on why this is required.

File: samples/bookinfo/src/ratings/ratings.js (hunk 1 above)
Comment (vadimeisenbergibm) [reply samples/bookinfo/src/ratings/ratings.js]: fixed in bb38dabf8956b3702ecb8922f0c0cf6e3d1a652d

File: samples/bookinfo/src/ratings/ratings.js (hunk 2 above)
Comment (vadimeisenbergibm) [reply samples/bookinfo/src/ratings/ratings.js]: Addressed in 6b4dac8e532d6b1ff5f56596ad99c33f2abf42ef

File: samples/bookinfo/src/ratings/ratings.js [hunk 3]
```diff
@@ -33,6 +35,35 @@ if (process.env.SERVICE_VERSION === 'v2') {
...
+
+  try {
+    ratings = JSON.parse(req.body)
+  } catch (error) {
+    res.writeHead(400, {'Content-type': 'application/json'})
+    res.end(JSON.stringify({error: 'please provide valid ratings JSON'}))
+    return
+  }
+
+
```
Comment (venilnoronha) [inline samples/bookinfo/src/ratings/ratings.js]: nit: remove extra line break

**Comment by vadimeisenbergibm** [comment]
@rshriram yep, I will publish new images.

**Comment by vadimeisenbergibm** [comment]
@wenchenglu Could you please merge this PR? It is an added functionality to bookinfo sample.

**Pull request merged**